projects/interfaces/buffer_test.go

```diff
@@ -0,0 +1,51 @@
+package OurBuffer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// If you make a buffer named b containing some bytes, calling b.Bytes() returns the same bytes you created it with.
+func TestBufferBytesReturnsInitialBytes(t *testing.T) {
+	inputBytes := []byte("Hello, World!")
+	b := NewBuffer(inputBytes)
+
+	bytes := b.Bytes()
+
+	require.Equal(t, len(inputBytes), len(bytes))
+}
+
+// If you write some extra bytes to that buffer using b.Write(), a call to b.Bytes() returns both the initial bytes and the extra bytes.
+func TestBufferWriteExtraBytes(t *testing.T) {
+	inputBytes := []byte("Hello ")
+	b := NewBuffer(inputBytes)
+	n, _ := b.Write([]byte("World!"))
```

Comment:
In general I'd recommend always doing the `require.NoError(t, err)`. We know our implementation _always_ returns `nil` which is worth verifying, but if not, seeing that there was an error is often much clearer to debug than seeing that the bytes aren't what you expect.